### ISSUE TITLE
[Bug]: Clarity needed for singularity deal send-manual - online, offline and commp

### ISSUE THREAD

**Issue by distora-w3** [data-preservation-programs/singularity#235]
### Description

```text
singularity deal send-manual is confusing

[issue 1] how to specify offline / online
[issue 2] template needs to be quoted otherwise interferes with rest of commandline
[issue 3] commp. Where calculated, how to calculate and how to specify on cli
[issue 4] command line arguments
```

### Steps to Reproduce

This is an example boost offline-deal that is clear and self-documenting.

```
boost offline-deal 
   --provider=f01943041  
   --commp=baga6ea4seaqavmm3asb5kejvgqel4vootsrzyosaaqs2fttbtsw7767auj6iyma 
   --car-size=252341037 
   --piece-size=268435456  
   --payload-cid=bafybeiclhef7cikojil5z2dr2gghgmvq2ecfzdbbvvwnpm7xq2pd2km7jm 
   --storage-price=0 
   --start-epoch=2947699 
   --verified=true 
   --wallet=f1nlhouj6cwctb4lijgh4tej7hhvdlwc5oc3byp5y 
   --duration=1526400
```
[Issue 1] in above example the online & offline is explicit

[Issue2] Unless quoted the following command fails

```
singularity deal send-manual --url-template http://45.32.x.x/ baga6ea4seaqfn2t5gxrg3by5i73cm2x6it7rz37fwsebxtwgqtjpatjkq2meoiy.car  f1nlhouj6cwctb4lijgh4tej7hhvdlwc5oc3byp5y  f01943041 baga6ea4seaqfn2t5gxrg3by5i73cm2x6it7rz37fwsebxtwgqtjpatjkq2meoiy 32GiB

ERROR: invalid parameter: client address not found

Bellow will pass through to boost
singularity deal send-manual --url-template "http://45.32.x.x/ baga6ea4seaqfn2t5gxrg3by5i73cm2x6it7rz37fwsebxtwgqtjpatjkq2meoiy.car"  f02130217    f01943041 baga6ea4seaqfn2t5gxrg3by5i73cm2x6it7rz37fwsebxtwgqtjpatjkq2meoiy 32GiB

```

[Issue 3] Commp is not obvious
- should we have Commp at singularity level?
- if yes, how should it be specified?
- if not specified boost will try locally, I received the following message

`Error: failed to verify CommP: failed to generate CommP: performing local commp: failed to get CARv2 reader: EOF`

[Issue 4] Easy to make command line argument errors

```
Current structure
singularity deal send-manual [command options] CLIENT_ADDRESS PROVIDER_ID PIECE_CID PIECE_SIZE
```
It would be better to force options for clarity  and self documentation.
`singularity deal send-manual [command options] --wallet xxxxx --provider yyyy  --piece-size xxxxxxx PIECE_CID`

### Version

singularity v0.2.47-1f0d416-dirty

### Operating System

Linux

### Database Backend

SQLite

**Comment by xinaxu** [comment]
> how to specify offline / online

online and offline are a bit overloaded concept here and is already confusing to lots of user. There is online/offline in boost and there is also online/offline with legacy market. Singularity needs to support both legacy and boost and since they work differently, we don't want to make commands like `send-boost-online`, `send-boost-offline`, `send-legacy-online`, `send-legacy-offline` and in the future there could be more. The difference between online and offline is whether the CAR file can be transfered over the wire during the deal flow. We have `--url-template` to specify the source of the CAR file. With this parameter, you're telling the SP that the CAR file can be retrieved from that URL directly and that implies online-deal without mentioning the online/offline concept. We could however, explain and categorize them nicely so it is clearer, i.e.
```
Deal file source
  --URL-template Specify the URL where the SP can download the CAR file as part of deal flow
```

> Unless quoted the following command fails

You need the quote because you have extra space in your URL. Quoting the command line arguments is a commonly known practice for CLI. Quote are interpreted at the shell layer and is irrelevant to the actual CLI implementation

> commp. Where calculated, how to calculate and how to specify on cli

The purpose of `send-manual` is to send deal using parameters prepared by other software so you're on your own to understand and calculate those parameters.

> command line arguments

Agree

**Comment by xinaxu** [comment]
Fixed in #221
